File: back_end_fg/WebApplication1/WebApplication1/Models/Auction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace WimsApiMKI.Models
{
    public class Auction
    {
        private int _id;
        private WimsUser _twitcher;
        private WimsUser _user;
        private int? _minAuction;
        private int? _maxAuction;
        private int _currentAuction;
        private DateTime? _auctionDate;
        private bool _auctionValidation;
        private int _idPrice;
        private PriceToWin _auctionPrice;
        private bool _active;

        public Auction()
        {
            Twitcher = new WimsUser();
            User = new WimsUser();
            AuctionPrice = new PriceToWin(_idPrice);
            AuctionDate = DateTime.Now;
        }

        public Auction(int value)
        {
            CurrentAuction = value;
        }

        #region GetSet
        public int Id
        {
            get { return _id; }
            set { _id = value; }
        }

        public WimsUser Twitcher
        {
            get { return _twitcher; }
            set { _twitcher = value; }
        }
        
        public WimsUser User
        {
            get { return _user; }
            set { _user = value; }
        }

        public int? MinAuction
        {
            get { return _minAuction; }
            set { _minAuction = value; }
        }

        public int? MaxAuction
        {
            get { return _maxAuction; }
            set { _maxAuction = value; }
        }
       
       public DateTime? AuctionDate
        {
            get { return _auctionDate; }
            set { _auctionDate = value; }
        }

        public int CurrentAuction
        {
            get { return _currentAuction; }
            set { _currentAuction = value; }
        }

        public bool AuctionValidation
        {
            get { return _auctionValidation; }
            set { _auctionValidation = value; }
        }

        public int IdPrice
        {
            get { return _idPrice; }
            set { _idPrice = value; }
        }

        public PriceToWin AuctionPrice
        {
            get { return _auctionPrice; }
            set { _auctionPrice = value; }
        }

        public bool Active
        {
            get { return _active; }
            set { _active = value; }
        }
        #endregion

    }
}

File: WIMS-front/src/app/Observables/Auction.ts
import {User} from './User';

export class Auction {
    get Id(): number {
        return this.id;
    }

    set Id(value: number) {
        this.id = value;
    }

    get User(): User {
        return this.user;
    }

    set User(value: User) {
        this.user = value;
    }

    get MinAuction(): number {
        return this.minAuction;
    }

    set MinAuction(value: number) {
        this.minAuction = value;
    }

    get MaxAuction(): number {
        return this.maxAuction;
    }

    set MaxAuction(value: number) {
        this.maxAuction = value;
    }

    get CurrentAuction(): number {
        return this.currentAuction;
    }

    set CurrentAuction(value: number) {
        this.currentAuction = value;
    }

    get AuctionDate(): string {
        return this.auctionDate;
    }

    set AuctionDate(value: string) {
        this.auctionDate = value;
    }

    get AuctionValidation(): boolean {
        return this.auctionValidation;
    }

    set AuctionValidation(value: boolean) {
        this.auctionValidation = value;
    }

    get IdPrice(): number {
        return this.idPrice;
    }

    set IdPrice(value: number) {
        this.idPrice = value;
    }
    private id: number;
    private minAuction: number;
    private maxAuction: number;
    private auctionDate: string;
    private auctionValidation: boolean;
    constructor(
        private user: User,
        private currentAuction: number,
        private idPrice: number
    ) {}

}
